pyannote/audio/pipelines/multilabel_detection.py

```diff
@@ -0,0 +1,186 @@
+    """Compute the mean Fscore over all labels
+
+    """
+
+    @classmethod
+    def metric_name(cls):
+        return "AVG[Labels]"
+
+    def __init__(self, labels: List[str], collar=0.0, skip_overlap=False,
+                 beta=1., parallel=False, **kwargs):
+        self.parallel = parallel
+        self.metric_name_ = self.metric_name()
+        self.components_ = set(self.metric_components())
+        self.reset()
+        self.collar = collar
+        self.skip_overlap = skip_overlap
+        self.beta = beta
+        self.labels = labels
+        self.submetrics: Dict[str, DetectionPrecisionRecallFMeasure] = {
+            label: DetectionPrecisionRecallFMeasure(collar=collar,
+                                                    skip_overlap=skip_overlap,
+                                                    beta=beta,
+                                                    **kwargs)
+            for label in self.labels
+        }
+
+    def reset(self):
+        super().reset()
+        for submetric in self.submetrics.values():
+            submetric.reset()
+
+    def compute_components(self, reference: Annotation, hypothesis: Annotation, uem=None, **kwargs):
+
+        details = self.init_components()
+        for label, submetric in self.submetrics.items():
+            details[label] = submetric(reference=reference.label_timeline(label),
+                                       hypothesis=hypothesis.label_timeline(label),
+                                       uem=uem,
+                                       **kwargs)
+        return details
+
+    def compute_metric(self, detail: Dict[str, float]):
+        return np.mean(detail.values())
+
+    def __abs__(self):
+        return np.mean([abs(submetric) for submetric in self.submetrics.values()])
+
+
+class MultilabelDetection(Pipeline):
+    """"""
+
+    def __init__(self,
+                 classes: Optional[List[SpeakerClass]] = None,
+                 segmentation: PipelineModel = "pyannote/vtc",
+                 fscore: bool = False,
+                 **inference_kwargs,
+                 ):
+
+        super().__init__()
+
+        self.segmentation = segmentation
+        self.fscore = fscore
+
+        # load model and send it to GPU (when available and not already on GPU)
+        model = get_model(segmentation)
+        if model.device.type == "cpu":
+            (segmentation_device,) = get_devices(needs=1)
+            model.to(segmentation_device)
+
+        self.labels = model.specifications.classes if classes is None else classes
+        self.segmentation_inference_ = Inference(model, **inference_kwargs)
+
+        self.binarize_hparams = ParamDict(**{
+            class_name: ParamDict(
+                onset=Uniform(0., 1.),
+                offset=Uniform(0., 1.),
+                min_duration_on=Uniform(0., 2.),
+                min_duration_off=Uniform(0., 2.),
+                pad_onset=Uniform(-1., 1.),
+                pad_offset=Uniform(-1., 1.)
```

Comment:
In relation with https://github.com/pyannote/pyannote-pipeline/pull/34,  this is a good use case for freezing only parts of `ParamDict`. `pad_onset` and `pad_offset` are seldom useful and it makes sense to reduce the dimension of the hyperparameter search space by freezing them to 0.

Reply:
Yes! I wondered if `pad_{off,on}set` was relevant or not, since you weren't using it in the VAD pipeline. I'll freeze that once it's freezable. (What I could also do is not parameterize it though?)